src/centengine/engine_impl.cc

```diff
@@ -514,10 +542,101 @@ grpc::Status engine_impl::DeleteAllServiceComments(
   return grpc::Status::OK;
 }
 
-grpc::Status engine_impl::ProcessServiceCheckResult(
+grpc::Status engine_impl::RemoveHostAcknowledgement(
     grpc::ServerContext* context __attribute__((unused)),
-    const Check* request,
-    CommandSuccess* response __attribute__((unused))) {
+    const HostIdentifier* request,
+    CommandSuccess* response) {
+  auto fn = std::packaged_task<int32_t(void)>([request]() -> int32_t {
+    std::shared_ptr<com::centreon::engine::host> temp_host;
+
+    switch (request->identifier_case()) {
+      case HostIdentifier::kName: {
+        auto it = host::hosts.find(request->name());
+        if (it != host::hosts.end())
+          temp_host = it->second;
+        if (temp_host == nullptr)
```

Comment:
This works but not as you could suppose.
See https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool
Please prefer: if (!temp_host)